encoding/mkv.md

```diff
@@ -0,0 +1,505 @@
+all attachments should be added here as needed with their subtitle tracks
+(if a font is used in `op.ass`, only add it to the `op.mkv` file).
+Adding a title and setting track languages
+should also be done at this point.
+
+<!-- TODO: Verify this behavior works on non-mpv players -->
+<!-- TODO: Mention issue with corrupted or missing subtitle tracks:
+    basically what happens in mpv is that it will show all tracks from
+    across all referenced segments causing subtitles from the external
+    segment to become track #2 and not appear without manual switching
+    I've noted this happening with mal-formatted .ass, empty subtitle files,
+    and somewhat when tracks are missing or mis-aligned but this missing thing
+    specifically needs more testing
+-->
+
+
+### Typical Episode
+
+The first episode in the example that contains both the OP and the ED
+is episode 2, so we will start there.
+
+In our directory, there should be plaintext chapter files
+generated by `octrim`.
+Using the MKVToolNix-GUI's *Chapter Editor* mode,
+we will open this file.
+(This can also be done from the command-line: `mkvtoolnix-gui --edit-chapters "ep02_chapters.txt"`.)
+
+In the example above,
+the *Chapter editor* GUI should at first look like this:
+
+![Chapter editor](images/mkv/ep02_chapters_txt_editor1.png)
+
+Ordered chapters require a start *and* an end timestamp,
+so we will need to generate the *End* column here.
+This can be accomplished by right-clicking the *Edition entry*,
+and selecting *Additional modifications*.
+Make sure only *Derive end timestamps from start timestamps* is ticked.
+
+![Derive end timestamps](images/mkv/derive_endtimes.png)
+
+After deriving the end timestamps,
+delete the last chapter,
+as it was a placeholder generated by `octrim`
+to determine the end timestamp of the last chapter.
+
+On the *Edition entry*, we can now tick *Ordered*:
+
+![Ticking ordered](images/mkv/chapter_editor_ordered_tick.png)
+
+At this point, if we were to mux these chapters into our episode 2 file,
+they will act like normal chapters,
+and the episode will play without the OP and ED.
+
+
+### Adding in our OP
+
+Before we add in the OP,
+we need to determine the timestamps needed for the timeline
+to reference our `op.mkv`.
+
+If we had chapters generated for the OP,
+we could simply copy over those timestamps
+(we will do this with our *playall* file later).
+Instead, we will simply use the number of frames in the entire op
+to determine the end timestamp.
+
+From our trim, `op = bdmv[30042:32201]`,
+we find that the clip is $$32201-30042=2159$$ frames long.
+This can also be found from `op.num_frames` in VapourSynth.
+
+Using a small [script to change frame numbers into timestamps][f2ts.py],
+we find that 2159 frames translates to `00:01:30.048291667`.
+This can also be accomplished with this following line in a Python instance.
+
+```py
+for _ in input().split():print('%02d:%02d:%09f'%((m:=(s:=int(_)/23.976)//60)//60,m%60,s%60))
```

Comment:
You should *never* use `_` as a variable name because that has special meaning in interactive sessions and is used as a "drop this" convention in scripts.